src/utils.ts

```diff
@@ -93,3 +93,7 @@ export const checkIfEnvironmentVariableIsOmitted = (key: string): boolean => {
 
     return omitEnvironmentVariablesWithPrefix.some((prefix: string) => key.toUpperCase().startsWith(prefix));
 }
+
+export const escapeDoubleQuote = (value: string): string => {
+    return value.replaceAll('"', '\\"');

```

Comment:
The current implementation uses double quotes to quote the environment variable:

https://github.com/Azure/cli/blob/68f0d36771d0cba0ebbf26993275a8e0b74a541f/src/main.ts#L57

However, only escaping double quotes may not be sufficient. [Command Substitution](https://www.gnu.org/software/bash/manual/bash.html#Command-Substitution) can happen in double quotes, which can also cause shell injection:

```
$ docker run -e "MY_KEY=$(whoami)" -it mcr.microsoft.com/azure-cli env
...
MY_KEY=user2
```

Note that `whoami` gets executed.

Using [single quotes](https://www.gnu.org/software/bash/manual/bash.html#Single-Quotes) to quote the environment variable is safer as this usage preserves the literal value of each character within the quotes.

However, this is still not the optimal solution as we need to care about single quotes' escaping. To make things more complicated:

> A single quote may not occur between single quotes, even when preceded by a backslash.
> \- https://www.gnu.org/software/bash/manual/bash.html#Single-Quotes

See how weird it becomes:

```py
# test.py
import shlex
print(shlex.quote("Single quote ' in string"))

# output
'Single quote '"'"' in string'
```

Is there a method in TypeScript that is similar to Python's `subprocess.Popen` which takes `args` as a list, instead of string?

Reply:
As it tested in [action test](https://github.com/MoChilia/ActionsDemo/actions/runs/5152677123), both environment variables with Command Substitution and single quote can pass the test. 
1. About Command Substitution: Not sure why Command Substitution didn't happen in the test, but it won't be a concern in this case. Since whether it happens or not, as long as there are no double quotes input in `docker run -e` which ensured by the function `escapeDoubleQuote`, the command won't fail.  
2. About using single quotes to quote the environment variables: It is not working in this situation. We have to use double quotes to quote the env.
3. About single quotes: Single quotes work well in this situation with no need to escape.
4. About taking args as a list: I agree with this great suggestion. It is a safer way to pass env. Will investigate on it.